flocker/control/test/test_httpapi.py

```diff
@@ -242,6 +242,76 @@ def test_container_name_collision_different_node(self):
         """
         return self._container_name_collision_test(self.NODE_A, self.NODE_B)
 
+    def test_create_container_with_environment(self):
+        """
+        A valid API request to create a container including environment
+        variables results in an updated configuration.
+        """
+        saving = self.persistence_service.save(Deployment(
```

Comment:
Oh, perhaps I understand. We're testing that *existing* configuration is updated rather than replaced, right?